packages/tailwind/src/tailwind.tsx

```diff
@@ -19,214 +12,127 @@ export interface TailwindProps {
-      ...(typeof resultingStyle === "undefined"
-        ? {}
-        : { style: resultingStyle }),
-    },
-    ...resultingChildren,
+  const nonInlineStylesToApply = sanitizedMediaQueries.filter(
+    (style) => style.trim().length > 0,
   );
 
-  // if this is a component, then we render it and recurse it through processElement
-  if (typeof modifiedElement.type === "function") {
-    const component = modifiedElement.type as React.FC;
-    const renderedComponent = component(modifiedElement.props);
-    if (React.isValidElement(renderedComponent)) {
-      modifiedElement = processElement(
-        renderedComponent,
-        nonMediaQueryTailwindStylesPerClass,
-        nonEscapedMediaQueryClasses,
+  function processElement(
+    element: React.ReactElement<EmailElementProps>,
+  ): React.ReactElement<EmailElementProps> {
+    const propsToOverwrite = {} as Partial<EmailElementProps>;
+
+    if (element.props.children) {
+      propsToOverwrite.children = React.Children.map(
+        element.props.children,
+        (child) => {
+          if (React.isValidElement<EmailElementProps>(child)) {
+            return processElement(child);
+          }
+
+          return child;
+        },
       );
     }
-  }
-
-  return modifiedElement;
-}
 
-type AnyElement = React.ReactElement<React.HTMLAttributes<HTMLElement>>;
-
-type HeadElement = React.ReactElement<
-  HeadProps,
-  string | React.JSXElementConstructor<HeadProps>
->;
-
-function processHead(
-  headElement: HeadElement,
-  responsiveStyles: string[],
-): React.ReactElement {
-  /*                   only minify here since it is the only place that is going to be in the DOM */
-  const styleElement = <style>{minifyCss(responsiveStyles.join(""))}</style>;
-
-  return React.cloneElement(
-    headElement,
-    headElement.props,
-    ...React.Children.toArray(headElement.props.children),
-    styleElement,
-  );
-}
-
-export const Tailwind: React.FC<TailwindProps> = ({ children, config }) => {
-  let headStyles: string[] = [];
-
-  const markupWithTailwindClasses = quickSafeRenderToString(<>{children}</>);
-  const markupCSS = useRgbNonSpacedSyntax(
-    getCssForMarkup(markupWithTailwindClasses, config),
-  );
+    if (element.props.className) {
+      const { styles, residualClassName } = inline(element.props.className);
+      propsToOverwrite.style = {
+        ...element.props.style,
+        ...styles,
+      };
+      if (residualClassName.trim().length > 0) {
+        propsToOverwrite.className = residualClassName;
+        /*
+          We sanitize only the class names of Tailwind classes that we are not going to inline
+          to avoid unpredictable behavior on the user's code. If we did sanitize all class names
+          a user-defined class could end up also being sanitized which would lead to unexpected 
+          behavior and bugs that are hard to track.
+        */
+        for (const singleClass of nonInlinableClasses) {
```

Comment:
This is the place where we use the returned `nonInlinableClasses` that I mentioned [here](https://github.com/resend/react-email/pull/1351/files#r1519508134). The comment pretty much sums up why we need to do this.

Currently the only place this is used.